### PR TITLE
performance-only fixes (np.clip, np.array growing, )

### PR DESCRIPTION
**Pull request by michaelfeil**
as discussed in #101:
- `np.clip` is slower than `np.core.umath.clip` (saves ~10%) simulation time
- lists instead of np.array growing and smaller fixes (saves ~10% simulation time)

Time before for 10 runs for pusing task (200 steps,3 resets) : `~13.0s +- 200ms `after `10.8s +- 150ms`

All changed functions have been like: 
`np.testing.assert_array_equal(prior_implementation, faster_implementation)` while running the pytests

### CODE REVIEW

**Comment by ftraeuble** [comment]
@michaelfeil Could you squash all the commits locally into one commit and force-push again to update the CR? thanks!

File: causal_world/task_generators/base_task.py [hunk 1]
```diff
@@ -880,48 +882,49 @@ def filter_structured_observations(self):
...
                         self._robot.\
-                            normalize_observation_for_key\
-                            (key=key, observation=
-                            self._non_default_robot_observation_funcs[key]())
-                    observations_filtered = \
-                        np.append(observations_filtered,
-                                  normalized_observations)
+                            normalize_observation_for_key(
+                                key=key, 
+                                observation=\
```
Comment (luator) [inline causal_world/task_generators/base_task.py:893]: Is the backslash needed here?

File: causal_world/task_generators/base_task.py [hunk 2]
```diff
@@ -880,48 +882,49 @@ def filter_structured_observations(self):
...
             else:
-                observations_filtered = \
-                    np.append(observations_filtered,
-                              np.array(self._current_full_observations_dict[key]))
+                new_obs = self._current_full_observations_dict[key]
+            
+            if type(new_obs) is np.ndarray:
+                observations_filtered.extend(new_obs.flat)
+            elif type(new_obs) is list or  type(new_obs) is tuple:
+                observations_filtered.extend(new_obs)
```
[the remark this answers is not in the document]
Comment (luator) [reply causal_world/task_generators/base_task.py:904]: Apart from this, wouldn't it be nicer to convert to a common type during assignment in the if/else above (when the actual type is known)?

**Review by luator — APPROVED** [review]
Thanks for the improvements!
I have some minor comments below.

**Review by michaelfeil — COMMENTED** [review]
https://github.com/rr-learning/CausalWorld/issues/103

File: causal_world/task_generators/base_task.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (luator) [reply causal_world/task_generators/base_task.py]: I see, I assumed that above the type is always known.  All good then.

**Pull request merged**